core/jvm/src/test/scala/zio/RTSSpec.scala

```diff
@@ -1209,6 +1212,20 @@ class RTSSpec(implicit ee: ExecutionEnv) extends TestRuntime {
       b      <- effect.await
     } yield b) must_=== 42
 
+  def testFirstSuccessOfValues =
+    unsafeRun(IO.firstSuccessOf(IO.fail(42), List(IO.succeed(24))).either) must_=== Right(24)
+
+  def testFirstSuccessOfFailures = {
+    implicit val d
+      : Diffable[Left[Int, Nothing]] = Diffable.eitherLeftDiffable[Int] //    TODO: Dotty has ambiguous implicits
+    unsafeRun(ZIO.firstSuccessOf(IO.fail(24).delay(10.millis), List(IO.fail(24))).either) must_=== Left(24)
```

Comment:
Maybe use a different value for the 2 failures, so that you check it returns the last one. Since both are 24 we can't be sure.

Reply:
Due to jitter, sometimes the `10.millis` will be first. It's best to use `IO.never` here for the left value.

Reply:
Will make the changes